File: TDDMicroExercises/Javascript/tire-pressure-monitoring-system/alarm.js

Alarm = function() {

	this._lowPressureTreshold = 17;
	this._highPressureTreshold = 21;
	this._sensor = new Sensor();
	this._alarmOn = false;
	this._alarmCount = 0;
};

Alarm.prototype = {

	check: function () {
		
		var psiPressureValue = this._sensor.popNextPressurePsiValue();

		if (psiPressureValue < this._lowPressureTreshold || this._highPressureTreshold < psiPressureValue)
		{
			this._alarmOn = true;
			this._alarmCount += 1;
		}
	},
	
	alarmOn: function () {
		 return this._alarmOn;
	}

};

File: TDDMicroExercises/Java/TirePressureMonitoringSystem/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Alarm.java
package tddmicroexercises.tirepressuremonitoringsystem;

public class Alarm implements IAlarm
{
    private final double LowPressureTreshold = 17;
    private final double HighPressureTreshold = 21;

    Sensor sensor = new Sensor();

    boolean alarmOn = false;
    private long alarmCount = 0;

    public void check()
    {
        double psiPressureValue = sensor.popNextPressurePsiValue();

        if (psiPressureValue < LowPressureTreshold || HighPressureTreshold < psiPressureValue)
        {
            alarmOn = true;
            alarmCount += 1;
        }
    }

    public boolean isAlarmOn()
    {
        return alarmOn; 
    }
}
